File: fixtures/scale/scala/large/src/cryptobench/store/JsonLines.scala
package cryptobench.store

object JsonLines {
  def toJson(record: ResultRecord): String = {
    val sb = new StringBuilder("{")
    field(sb, "suite", "\"" + record.suite + "\"")
    field(sb, "passed", record.passed.toString)
    field(sb, "failed", record.failed.toString)
    field(sb, "millis", String.format("%.3f", Double.box(record.millis)))
    sb.setLength(sb.length - 1)
    sb.append('}').toString
  }

  private def field(sb: StringBuilder, key: String, value: String): Unit = {
    sb.append('"').append(key).append('"').append(':').append(value).append(',')
    ()
  }
}

File: fixtures/scale/dart/large/lib/store/json_lines.dart
import 'result_record.dart';

String toJsonLine(ResultRecord record) {
  final sb = StringBuffer('{');
  _field(sb, 'suite', '"${record.suite}"');
  _field(sb, 'passed', record.passed.toString());
  _field(sb, 'failed', record.failed.toString());
  _field(sb, 'millis', record.millis.toStringAsFixed(3));
  final text = sb.toString();
  return '${text.substring(0, text.length - 1)}}';
}

void _field(StringBuffer sb, String key, String value) {
  sb.write('"$key":$value,');
}
